module/zcommon/zfeature_common.c

```diff
@@ -247,19 +247,13 @@ zfs_mod_list_supported_free(struct zfs_mod_supported_features *list)
 static boolean_t
 zfs_mod_supported_impl(const char *scope, const char *name, const char *sysfs)
 {
-	boolean_t supported = B_FALSE;
-	char *path;
-
-	int len = asprintf(&path, "%s%s%s%s%s", sysfs,
-	    scope == NULL ? "" : "/", scope == NULL ? "" : scope,
-	    name == NULL ? "" : "/", name == NULL ? "" : name);
-	if (len > 0) {
-		struct stat64 statbuf;
-		supported = !!(stat64(path, &statbuf) == 0);
-		free(path);
-	}
-
-	return (supported);
+	char path[128];
+	if (snprintf(path, sizeof (path), "%s%s%s%s%s", sysfs,
+	    scope == NULL ? "" : "/", scope ?: "",
+	    name == NULL ? "" : "/", name ?: "") != sizeof (path))
```

Comment:
Should snprintf() return value be checked for -1 or >= sizeof (path) to cover error/truncate cases?

Reply:
`snprintf()` won't return -1 because it can't encounter a an output error. As for the bound check: you're right, it was wrong, it should've been `< sizeof(path)`.